scalyr_agent/builtin_monitors/docker_monitor.py

```diff
@@ -763,6 +763,20 @@ def _get_containers(
                                     info["LogPath"] if "LogPath" in info else None
                                 )
 
+                                if not log_path:
+                                    # NOTE: If docker_raw_logs is True and we hit this code path it
+                                    # really means we won't be ingesting any logs so this should
+                                    # really be treated as a fatal error.
+                                    logger.warning(
```

Comment:
An unrelated issue I encountered when trying to reproduce the original issue.

``LogPath`` being None will in 95% of cases represent a misconfiguration on the Docker daemon side - e.g. the daemon is using some other log-driver such as ``jourdnald`` and not ``json-file`` one.

It could likely also happen under some other scenarios, but it basically represents a fatal error if ``docker_raw_logs`` is ``True`` (aka default value), since it means we won't be monitoring / ingesting logs for that container.

Right now we simply silently ignore this issue which is obviously bad and a nightmare to troubleshoot.

We can't really throw here (even if it most likely represents a configuration error), but I think we should at the very least log a message under error level and be very explicit - e.g. something along the lines of "LogPath for container foo is None. This means we won't be monitoring and ingesting logs for this container if docker_raw_logs mode is used".

What do other people think?